CMakeLists.txt

```diff
@@ -1,5 +1,14 @@
 cmake_minimum_required(VERSION 3.13.4)
 
+#-----------------------------------------------------------------------------
+# CCache
+#-----------------------------------------------------------------------------
+option(Slicer_USE_CCACHE "Build ${PROJECT_NAME} using CCache." OFF)
+find_program(CCACHE_PROGRAM ccache)
+if(CCACHE_PROGRAM)
+    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CCACHE_PROGRAM}")
```

Comment:
This would compete with https://github.com/Kitware/CMake/blob/4fd2d62613af072204910b303fc1506804279df1/Modules/CTestUseLaunchers.cmake#L70